mithril-explorer/src/components/CertifyCardanoTransactionsModal/index.js

```diff
@@ -0,0 +1,211 @@
+import { MithrilClient } from "@mithril-dev/mithril-client-wasm";
+import React, { useEffect, useState } from "react";
+import { Alert, Col, Modal, Row, Tab } from "react-bootstrap";
+import { useSelector } from "react-redux";
+import { fetchGenesisVerificationKey } from "../../utils";
+import CertificateVerifier, {
+  certificateValidationSteps,
+} from "../VerifyCertificate/CertificateVerifier";
+import TransactionCertificationBreadcrumb from "./TransactionCertificationBreadcrumb";
+import TransactionCertificationResult from "./TransactionCertificationResult";
+import FetchingProofPane from "./FetchingProofPane";
+import ValidatingProofPane from "./ValidatingProofPane";
+
+export const validationSteps = {
+  ready: 1,
+  fetchingProof: 2,
+  validatingCertificateChain: 3,
+  validatingProof: 4,
+  done: 5,
+};
+
+export default function CertifyCardanoTransactionsModal({
+  transactionHashes,
+  onHashesChange = (hash) => {},
+}) {
+  const currentAggregator = useSelector((state) => state.settings.selectedAggregator);
+  const [certificate, setCertificate] = useState(undefined);
+  const [certificateVerifierStep, setCertificateVerifierStep] = useState(
+    certificateValidationSteps.ready,
+  );
+  const [transactionsProofs, setTransactionsProofs] = useState({});
+  const [showLoadingWarning, setShowLoadingWarning] = useState(false);
+  const [isEverythingValid, setIsEverythingValid] = useState(false);
```

Comment:
Maybe something like this to be more precise:
```suggestion
  const [isMessageMatchCertificate, setIsMessageMatchCertificate] = useState(false);
```

Reply:
I agree that it could be worded better but in this case there's two checks that are encapsulated by this boolean: 
1. That, as you said, the proof does match the certificate it claims to.
2. That the proof in itself is a valid MerkleProof.

So you're proposal won't do the trick here :/.

Reply:
It's not perfect but I renamed it to `isProofValid`.